modules/3d/test/ocl/test_tsdf.cpp

```diff
@@ -571,6 +571,282 @@ void valid_points_test_common_framesize(VolumeType volumeType, VolumeTestSrcType
+    double nrmDiffNorm = cv::sum(maskNrmDiff)[0]/255.0;
+
+    EXPECT_LE(ptsDiffNorm, 786);
+    EXPECT_LE(nrmDiffNorm, 786);
+
+    double normPts = cv::norm(mpts, mptsRot, NORM_INF, (maskPts0 & maskPtsRot));
+    Mat absdot = normalsError(mnrm, mnrmRot);
+    double normNrm = cv::norm(absdot, NORM_L2, (maskNrm0 & maskNrmRot));
+
+    EXPECT_LE(normPts, 2.0);
+    EXPECT_LE(normNrm, 73.05);
+}
+
+
+void debugVolumeDraw(const Volume &volume, Affine3f pose, Mat depth, float depthFactor, std::string objFname)
+{
+    Vec3f lightPose = Vec3f::all(0.f);
+    UMat points, normals;
+    volume.raycast(pose.matrix, points, normals);
+
+    UMat ptsList, ptsList3, nrmList, nrmList3;
+    volume.fetchPointsNormals(ptsList, nrmList);
+    // transform 4 channels to 3 channels
+    cvtColor(ptsList, ptsList3, COLOR_BGRA2BGR);
+    cvtColor(ptsList, nrmList3, COLOR_BGRA2BGR);
+    savePointCloud(objFname, ptsList3, nrmList3);
+
+    displayImage(depth, points.getMat(ACCESS_READ), normals.getMat(ACCESS_READ), depthFactor, lightPose);
+}
+
+
+void boundingBoxGrowthTest(VolumeType volumeType)
+{
+    VolumeSettings vs(volumeType);
+    Volume volume(volumeType, vs);
+
+    if ( volumeType == VolumeType::TSDF || volumeType == VolumeType::ColorTSDF )
+    {
+        Vec3i res;
+        vs.getVolumeResolution(res);
+        float voxelSize = vs.getVoxelSize();
+        Matx44f pose;
+        vs.getVolumePose(pose);
+        Vec3f end   = voxelSize * Vec3f(res);
+        Vec6f truebb(0, 0, 0, end[0], end[1], end[2]);
+        Vec6f bb = volume.getBoundingBox(Volume::BoundingBoxPrecision::VOLUME_UNIT);
+        Vec6f diff = bb - truebb;
+        double normdiff = std::sqrt(diff.ddot(diff));
+        ASSERT_LE(normdiff, std::numeric_limits<double>::epsilon());
+    }
+    else // HashTSDF
+    {
+        Size frameSize(vs.getRaycastWidth(), vs.getRaycastHeight());
+        Matx33f intr;
+        vs.getCameraIntegrateIntrinsics(intr);
+        bool onlySemisphere = false;
+        float depthFactor = vs.getDepthFactor();
+        Ptr<Scene> scene = Scene::create(frameSize, intr, depthFactor, onlySemisphere);
+        std::vector<Affine3f> poses = scene->getPoses();
+
+        Mat depth = scene->depth(poses[0]);
+        UMat udepth;
+        depth.copyTo(udepth);
+
+        // depth is integrated with multiple weight
+        //TODO: add weight parameter to integrate() call (both scalar and array of 8u/32f)
+        const int nIntegrations = 1;
+        for (int i = 0; i < nIntegrations; i++)
+            volume.integrate(udepth, poses[0].matrix);
+
+        Vec6f bb = volume.getBoundingBox(Volume::BoundingBoxPrecision::VOLUME_UNIT);
+        Vec6f truebb(-0.9375f, 1.3125f, -0.8906f, 3.9375f, 2.6133f, 1.4004f);
+        Vec6f diff = bb - truebb;
+        double bbnorm = std::sqrt(diff.ddot(diff));
+        // it's OK to have such big difference since this is volume unit size-grained BB calculation
+        EXPECT_LE(bbnorm, 0.2228);
```

Comment:
I propose to compare the result with volume unit size or add volume unit size to the comment for reference.